asapdiscovery-docking/asapdiscovery/docking/workflows/workflows.py

```diff
@@ -0,0 +1,129 @@
+"""
+Schema for workflows base classes
+"""
+import logging
+from pathlib import Path
+from typing import Optional
+
+from asapdiscovery.data.dask_utils import DaskType
+from asapdiscovery.data.postera.manifold_data_validation import TargetTags
+from asapdiscovery.modeling.protein_prep_v2 import CacheType
+from pydantic import BaseModel, Field, PositiveInt, root_validator, validator
+
+
+class WorkflowInputsBase(BaseModel):
+    filename: Optional[str] = Field(
+        None, description="Path to a molecule file containing query ligands."
+    )
+
+    pdb_file: Optional[Path] = Field(
+        None, description="Path to a PDB file to prep and dock to."
+    )
+
+    fragalysis_dir: Optional[Path] = Field(
+        None, description="Path to a directory containing a Fragalysis dump."
+    )
+    structure_dir: Optional[Path] = Field(
+        None,
+        description="Path to a directory containing structures to dock instead of a full fragalysis database.",
+    )
+    postera: bool = Field(
+        False, description="Whether to use the Postera database as the query set."
+    )
+
+    cache_dir: Optional[str] = Field(
+        None, description="Path to a directory where a cache has been generated"
+    )
+
+    gen_cache: Optional[str] = Field(
+        None,
+        description="Generate a cache from structures prepped in this workflow run in this directory",
+    )
+
+    cache_type: Optional[list[str]] = Field(
+        [CacheType.DesignUnit], description="The types of cache to use."
+    )
+
+    target: TargetTags = Field(None, description="The target to dock against.")
+
+    write_final_sdf: bool = Field(
+        default=True,
+        description="Whether to write the final docked poses to an SDF file.",
+    )
+    use_dask: bool = Field(True, description="Whether to use dask for parallelism.")
+
+    dask_type: DaskType = Field(
+        DaskType.LOCAL, description="Dask client to use for parallelism."
+    )
+
+    dask_cluster_n_workers: PositiveInt = Field(
+        10,
+        description="Number of workers to use as inital guess for Lilac dask cluster",
+    )
+
+    dask_cluster_max_workers: PositiveInt = Field(
+        200, description="Maximum number of workers to use for Lilac dask cluster"
+    )
+
+    n_select: PositiveInt = Field(
+        5, description="Number of targets to dock each ligand against."
+    )
+
+    # TODO: make this an "abstractmethod" kinda thing that needs to be re-named in subclasses
```

Comment:
I think you can just leave out `"docking_workflow_base"` as default param and the model can't be instantiated.

eg 

```python
    logname: str = Field(..., description="Name of the log file.")


```

Reply:
ahhh i didn't know you could do that, that's cool